packages/firefox.rb

```diff
@@ -100,4 +100,18 @@ def self.postinstall
     end
     ExitMessage.add "\nType 'firefox' to get started.\n"
   end
+
+  def self.postremove
+    Dir.chdir("#{CREW_PREFIX}/bin") do
+      if File.symlink?('x-www-browser') \
+        && (File.realpath('x-www-browser') == "#{CREW_PREFIX}/bin/firefox")
```

Comment:
My fault 😅 Doesn't know `File.realpath` will fail on invalid symlink. Use `File.readlink` instead?

Reply:
That didn't work.  I think the main issue here is we should be using `self.preremove` instead of `self.postremove` because in the postremove stage, the firefox executable has already been removed.